casbin/abac_data.h

```diff
@@ -0,0 +1,111 @@
+/*
+* Copyright 2021 The casbin Authors. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#ifndef ABAC_H
+#define ABAC_H
+
+#include <unordered_map>
+#include <string>
+#include <vector>
+#include <variant>
+#include <memory>
+
+namespace casbin {
+
+/**
+ * @brief A wrapper to contain ABAC entity with a list of attributes stored in a hashmap
+ * 
+ */
+class ABACData {
+
+public:
+// Array containing the reference to instantiated ABACData so far
+static std::vector<std::shared_ptr<ABACData>> s_dataSet;
+
+private:
+
+    // Intrinsic definitions
+    typedef std::variant<std::string, int32_t, float> VariantType;
+    typedef std::unordered_map<std::string, VariantType> VariantMap;
```

Comment:
This may also be:

```suggestion
    typedef std::vector<std::pair<std::string, VariantType>> VariantList;
```

Let me know which one is better.

Reply:
I think `unordered_map` is better.